### PR TITLE
daemon: rename local server to daemon server

### PR DESCRIPTION
**Pull request by dbentley**

### CODE REVIEW

File: daemon/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Scoot Daemon Server.
+
+A server that serves the Scoot Deamon API.
+
+For now this is across Unix Domain Sockets, but this detail is hidden behind the interfaces.
```
Comment (jasonbs10) [inline daemon/README.md]: User has to manually create ~/.scoot, so not quite hidden. Can you mkdir?

File: daemon/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Scoot Daemon Server.
+
+A server that serves the Scoot Deamon API.
```
Comment (jasonbs10) [inline daemon/README.md]: sp.

**Comment by jasonbs10** [comment]
lgtm

File: daemon/README.md (hunk 1 above)
Comment (CaitieM20) [inline daemon/README.md]: Nit: typo Daemon not Deamon

Comment (dbentley) [inline daemon/README.md]: I'd like to keep this the minimal changes for the rename, so I added TODO's here and in locate.go.

File: daemon/README.md (hunk 2 above)
Comment (dbentley) [inline daemon/README.md]: Fixed.

**Comment by CaitieM20** [comment]
LGTM

**Pull request merged**